projects/scream/scream_post/translate_prognostic_output.py

```diff
@@ -0,0 +1,325 @@
+}
+
+RENAME_ML_CORRECTION_TO_MACHINE_LEARNING = {
+    "mlcorrection_T_mid_tend": "machine_learning_T_mid_tend",
+    "mlcorrection_qv_tend": "machine_learning_qv_tend",
+    "mlcorrection_horiz_winds_tend": "machine_learning_horiz_winds_tend",
+}
+
+TEMP = "air_temperature"
+SPHUM = "specific_humidity"
+DELP = "pressure_thickness_of_atmospheric_layer"
+EASTWARD_WIND = "eastward_wind"
+NORTHWARD_WIND = "northward_wind"
+
+
+def _get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "input_data",
+        type=str,
+        default=None,
+        help="Input netcdf file(s) in string, wildcards allowed.",
+    )
+    parser.add_argument(
+        "output_path",
+        type=str,
+        help="Local or remote path where output will be written.",
+    )
+    parser.add_argument(
+        "chunk_size",
+        type=int,
+        help="Chunk size for the time dimension of output zarrs.",
+    )
+    parser.add_argument("variable_category", type=str, help="Either 2d, 3d, or all.")
+    parser.add_argument(
+        "--subset",
+        help="Whether to subset to a smaller set of timesteps",
+        action="store_true",
+    )
+    return parser
+
+
+def compute_PRATEsfc(ds):
+    """
+    Compute PRATEsfc from precip_liq_surf_mass_flux and precip_ice_surf_mass_flux
+    """
+    if "PrecipIceSurfMassFlux" in ds.variables:
+        ds = ds.rename({"PrecipIceSurfMassFlux": "precip_liq_surf_mass_flux"})
+    if "PrecipLiqSurfMassFlux" in ds.variables:
+        ds = ds.rename({"PrecipLiqSurfMassFlux": "precip_ice_surf_mass_flux"})
+    # scream native output saves "mass flux" as m s^-1
+    # we multiply by the water density (1000) to get to mass flux (kg m^-2 s^-1)
+    PRATEsfc = (ds.precip_liq_surf_mass_flux + ds.precip_ice_surf_mass_flux) * 1000.0
+    PRATEsfc = PRATEsfc.assign_attrs(units="kg/m^2/s")
+    ds["PRATEsfc"] = PRATEsfc
+    return ds
+
+
+def add_additional_surface_fields(ds):
+    """
+    Add shortwave_transmissivity_of_atmospheric_column
+    and override_for_time_adjusted_total_sky_downward_longwave_flux_at_surface
+    for training ML radiaitve fluxes
+    Add PRATEsfc for training diagnostic purposes
+    """
+    ds = add_rad_fluxes(ds)
+    ds = compute_PRATEsfc(ds)
+    return ds
+
+
+def determine_nudging_or_ML_run(ds):
+    if "nudging_T_mid_tend" in ds.variables:
+        return "nudging"
+    elif "machine_learning_T_mid_tend" in ds.variables:
+        return "machine_learning"
+    else:
+        return None
+
+
+def compute_prognostic_run_diagnostics(
+    ds: xr.DataArray, label: str, tendency_variables: [], hydrostatic: bool = False
+):
+    """
+    Compute diagnostics for a prognostic run
+
+    Args:
+        ds: xarray dataset containing prognostic run output
+        label: either "machine_learning" or "nudging"
+        tendency_variables: list of variables that are either nudged or ML corrected
+        hydrostatic: whether the run is hydrostatic
```

Comment:
Is it worth exposing this in the command-line interface?  Right now I guess it is always `False`.  I'm recognizing now that a lot of this is a product of symmetry with functions in [`prognostic_c48_run/runtime/diagnostics/compute.py`](https://github.com/ai2cm/fv3net/blob/master/workflows/prognostic_c48_run/runtime/diagnostics/compute.py), which seems reasonable to keep.

Reply:
Sticking with the same interface for consistency.